### ISSUE TITLE
SPRINT-2

### ISSUE THREAD

**Issue by rodrigonevesofi** [rodrigonevesofi/ML-CHATBOT-BICEPS#1]

**Comment by laraa-n** [comment]
Atividade 04 -
Presentes:
Rodrigo
Nicole
Barbara
Mateus
Letícia
Anderson

- Resultado da atividade

Intercepto (Tempo Base): 5.05 min
Coeficiente (Aumento por Caractere): 0.49 min

Previsão para 200 caracteres: 103.91 minutos

Amostra dos dados:
   tamanho_msg  tempo_real_espera
0          152          80.061051
1          229         120.585120
2          142          75.073165
3           64          36.068540
4          156          83.483925
/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(

**Comment by rodrigonevesofi** [comment]
**Aula 05 - Sprint 2**
----------------------------------------------
**Presentes:**
Mateus 
Nicole
Rodrigo 

**Atividade 05 - Resultado**
----------------------------------------------------

3. Média de caracteres: 52.70
4. Mediana de caracteres: 40.50
5. Moda de caracteres: 30
6. Desvio Padrão: 35.31

**Explicação do Resultado**

Os resultados mostram algumas informações sobre o tamanho das mensagens enviadas pelos usuários.

A média foi 52,70 caracteres. A média é calculada somando todos os valores e dividindo pela quantidade total de dados. Nesse caso, significa que o tamanho médio das mensagens enviadas pelos usuários é de aproximadamente 53 caracteres.

A mediana foi 40,50 caracteres. A mediana representa o valor que fica exatamente no meio quando todos os dados são colocados em ordem. Isso mostra que metade das mensagens tem menos de 40,5 caracteres e a outra metade tem mais. Como a mediana é menor que a média, isso sugere que existem algumas mensagens muito grandes que aumentam o valor da média.

A moda foi 30 caracteres. A moda é o valor que mais aparece no conjunto de dados. Isso mostra que o tamanho de mensagem mais comum entre os usuários é de cerca de 30 caracteres.

Por fim, o desvio padrão foi 35,31. O desvio padrão mede o quanto os valores se afastam da média. Esse valor relativamente alto indica que existe bastante variação no tamanho das mensagens, ou seja, alguns usuários enviam mensagens curtas enquanto outros enviam mensagens bem maiores.

De forma geral, os resultados mostram que a maioria das mensagens é relativamente curta, mas existem algumas mensagens maiores que aumentam a média e geram uma grande variação no tamanho das mensagens enviadas.

**Comment by leecampos2021-hash** [comment]
AULA 06 - SPRINT 2 
-----------------------------------------------
**Presentes:**
Rodrigo
Nicole
Barbara
Mateus
Letícia
Anderson

Atividade 01 - Resultado:
-----------------------------------
[nltk_data] Downloading package punkt_tab to /root/nltk_data...
['Machine', 'Learning', 'é', 'uma', 'das', 'áreas', 'mais', 'importantes', 'da', 'Inteligência', 'Artificial', '.']
A quantidade de tokens na mensagem é: 12
[nltk_data]   Unzipping tokenizers/punkt_tab.zip.

**Comment by laraa-n** [comment]
AULA 06 - SPRINT 2
----------------------------------------------------------------
**Presentes:**
Anderson
Barbara
Leticia
Mateus
Nicole
Rodrigo

Atividade 02 - Resultado e explicacao
------------------------------------------------------------------
**Resultado:**

--- Exercício 1: Normalizador ---
Original: ERRO Crítico no Servidor!!! Código: #404... favor verificar AGORA.
Limpo:    erro crítico no servidor código 404 favor verificar agora

--- Exercício 2: Filtro de Relevância ---
Frase:  O meu boleto ainda não chegou no meu e-mail
Tokens: ['boleto', 'ainda', 'chegou']
Frase:  Gostaria de falar com um atendente humano por favor
Tokens: ['gostaria', 'falar', 'atendente', 'humano', 'favor']

--- Exercício 3: Detector de Urgência ---
[ALTA PRIORIDADE] Mensagem: Olá, estou com um problema urgente, meu sistema está parado!
[NORMAL] Mensagem: Bom dia, gostaria de saber o preço do plano premium.
[ALTA PRIORIDADE] Mensagem: Erro fatal na integração, socorro!!!
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!

**Explicação do Resultado**

No Exercício 1, ocorre o processo de normalização textual, no qual o código converte todo o texto para letras minúsculas e remove sinais de pontuação, como “!”, “:”, “#”, entre outros. Esse processo é importante porque elimina elementos que não contribuem para o significado da frase e ajuda a padronizar o texto, facilitando as próximas etapas da análise.

No Exercício 2, o código realiza a tokenização, dividindo as frases em palavras menores chamadas tokens. Em seguida, ele remove as stopwords, que são palavras muito comuns (como “o”, “de”, “com”) e que não agregam muito valor ao entendimento da frase. Como resultado, permanecem apenas as palavras mais importantes, o que torna a análise mais eficiente.

No Exercício 3, o código funciona como um detector de urgência. Ele limpa o texto deixando em minúsculo e sem pontuação e verifica se existem palavras que indicam urgência, como “erro”, “problema” e “urgente”. Quando essas palavras aparecem, a mensagem é classificada como alta prioridade; caso contrário, é considerada normal. Essa etapa mostra como regras simples podem ser usadas para automatizar a classificação de mensagens.

**Comment by barbaraleao-28** [comment]
**AULA 06 - SPRINT 2**
-----------------------------------------------
**Presentes:**
Anderson
Barbara
Leticia
Mateus
Nicole
Rodrigo

Atividade 03 - Resultado e explicacao
-------------------------------------------------------------------

['quero', 'aprender', 'Machine', 'Learning', 'criar', 'chatbots', 'inteligentes']
[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!

- Explicação: 
No código disponibilizado, se te a biblioteca NLTK para executar o processo básico de Processamento de Linguagem Natural (PNL) em uma frase. No primeiro momento, ele vai importar as funções necessárias e garantir que a base de stopwords (palavras comuns do idioma) esteja disponível. Em seguida, ele vai definir a frase “Eu quero aprender Machine Learning para criar chatbots inteligentes” como a entrada. Depois no próximo passo, temos a tokenização, feita com word_tokenize, que divide a frase em uma lista de palavras individuais (tokens), como por exemplo “Eu”, “quero”, “aprender”, etc. Por fim, o código carrega uma lista de stopwords em português — palavras como “eu” e “para”, que são muito frequentes, mas pouco relevantes para o significado principal da frase.

Durante a etapa principal, o código percorre cada palavra da lista de tokens e verifica se ela, convertida para minúsculo com lower(), não está na lista de stopwords, isso é importante para garantir que palavras como “Eu” sejam corretamente reconhecidas como “eu” e removidas. Assim, as palavras que passam por esse filtro são adicionadas a uma nova lista chamada palavras_filtradas. No final, o programa imprime essa lista, que contém apenas os termos mais relevantes da frase: “quero”, “aprender”, “Machine”, “Learning”, “criar”, “chatbots” e “inteligentes”.

Resumidamente, o código vai realizar três etapas principais de PNL: separação  do texto em palavras, remoção de termos irrelevantes, mantendo apenas as palavras-chave, o que permite identificar a ideia central da frase, neste caso, o interesse em aprender Machine Learning para criar chatbots inteligentes.

**Comment by leecampos2021-hash** [comment]
ATIVIDADE 04 
--------------------------------------------
**Presentes:**
Rodrigo
Nicole
Barbara
Mateus
Letícia
Anderson

Resultado: 
-------------------------------------------------------------
['cancelar' 'comprar' 'eu' 'horário' 'qual' 'quero']
[[0 1 1 0 0 1]
 [1 0 1 0 0 1]
 [0 0 0 1 1 0]]


Nesse código, importamos a biblioteca sklearn.feature_extraction.text para utilizar o CountVectorizer, que realiza a transformação de textos em dados numéricos.

Em seguida, criamos uma lista com três frases que servirão como base para análise.

Depois, instanciamos o CountVectorizer, que será responsável por aprender o vocabulário presente nas frases.

O método fit_transform é utilizado para aprender todas as palavras únicas (vocabulário) e transformar as frases em uma matriz numérica baseada na contagem de palavras.

O método get_feature_names_out() exibe o vocabulário identificado.

Por fim, X.toarray() mostra a matriz numérica, onde cada linha representa uma frase e cada coluna representa a frequência de uma palavra no vocabulário.

**Comment by mateushg22** [comment]
**Aula 06 - Sprint 2**
-----------------------------------
**Presentes:**
Mateus
Anderson
Barbara
Rodrigo
Leticia
Nicole

Atividade 05
-------------------------------
frase = "eu gosto de aprender Machine Learning"

lista = frase.split()

total = len(lista)

print(total)

resultado = 6

A primeira linha eu crio a variável junto com texto  , já no de baixo ele esta dividindo a frase em palavras e contando um numero total de palavras tendo como resultado final  e 6 .

**Exemplo:** "eu"-1 ,"gosto"-2 ,"de"-3,"aprender"-4,"Machine"-5,"Learning"-6.

(Splint) = ele dividi um texto em partes, cada pedaço vira um item em uma lista.
(Len)= ele conta quantas palavras existe.

**Comment by laraa-n** [comment]
**Atividade 08** 
------------------------------------
**Presentes:**
Anderson
Barbara
Leticia
Mateus
Nicole
Rodrigo

Resultado:
------------------------------------------------------
 1 . Precisão do modelo 
Acurácia do modelo: 0.67
Interpretação: A acurácia de 0.67 indica que o modelo acerta aproximadamente 67% das previsões, ou seja, cerca de 7 a cada 10 mensagens são classificadas corretamente. Oque é razoável, mas não tão alto. Pode ocorrer pelo fato da base de dados ser pequena e não ter muitas variações de frases 

2.Funcionamento do Chatbot:

Acurácia do modelo: 0.67

=== CHATBOT INTELIGENTE ===
Digite 'sair' para encerrar

Usuário: quero cancelar meu pedido
Chatbot (cancelamento): Posso te ajudar a cancelar seu pedido.

Usuário: meu produto veio quebrado
Chatbot (reclamacao): Sinto muito pelo problema. Vamos resolver isso.

Usuário: qual o prazo de entrega?
Chatbot (consulta): Vou verificar essa informação para você.

Usuário: sair
Chatbot: Encerrando atendimento...


3. Testes:
Acurácia do modelo: 0.67

=== CHATBOT INTELIGENTE ===
Digite 'sair' para encerrar

Usuário: consulta 
Chatbot (cancelamento): Posso te ajudar a cancelar seu pedido.

Usuário: minha entrega atrasou
Chatbot (reclamacao): Sinto muito pelo problema. Vamos resolver isso.

Usuário: gostaria de revisar meu pedido
Chatbot (cancelamento): Posso te ajudar a cancelar seu pedido.

Usuário: qual o andamento do meu pedido?
Chatbot (consulta): Vou verificar essa informação para você.

Usuário: veio faltando algo no pedido
Chatbot (cancelamento): Posso te ajudar a cancelar seu pedido.

Usuário: meu pedido chegou errado
Chatbot (cancelamento): Posso te ajudar a cancelar seu pedido.

Usuário: sair
Chatbot: Encerrando atendimento...


**Avaliação do resultado:**
Não realizamos apenas os testes iniciais, pois identificamos alguns erros relacionados às requisições feitas pelo usuário. Portanto, os resultados não foram totalmente satisfatórios.

**Comment by rodrigonevesofi** [comment]
Atividade 06
--------------------------------------------------------------
**Presentes:**

Anderson
Barbara
Leticia
Mateus
Nicole
Rodrigo

Resultado 
----------------------------------------------------------------

Counter({'chatbot': 3, 'inteligência': 1, 'artificial': 1, 'aprendizado': 1})

texto = "chatbot chatbot inteligência artificial chatbot aprendizado" | Esse é o nosso texto base

palavras = texto.split() | O metodo utilizado "split" separa os textos pelos espaços

frequencia = Counter(palavras) | O "Counter" serve para fazer a contagem de quantas vezes cada palavra aparece

print(frequencia) | Exibe o resultado, mostrando quantas vezes cada palavra aparece no texto

**Comment by rodrigonevesofi** [comment]
Atividade 07
-
Presentes:

Anderson
Barbara
Leticia
Mateus
Nicole
Rodrigo

Resultado
--
Digite sua mensagem: vocês abrem no domingo
Intenção detectada: horario

frases = [...] - intencoes = [...] | Os dois basicamente criaram uma lista, listando as frases e suas intenções.

O CountVectorizer transforma texto em números, e cria uma matriz onde cada palavra vira uma coluna.

MultinomialNB é um algoritmo de "Naive Bayes"

Ele aprende os padrões das listas, entende suas intenções. E onde o treinamento da IA acontence.

predicao = modelo.predict(entrada_vetor)

O modelo analisa a frase e retorna a intenção mais provável.

O print é a saída/resultado.

O código cria um classificador de intenções usando o PNL. As frases são transformadas em vetores numéricos com o CountVectorizer, ele é um modelo "Naive Bayes" e é treinado para aprender padrões entre palavras e intenções. Quando o usuário digita uma nova frase, ela é convertida em vetor e o modelo prevê a intenção correspondente.
